server/src/main/java/org/elasticsearch/action/IndicesRequest.java

```diff
@@ -40,6 +40,13 @@
      */
     IndicesOptions indicesOptions();
 
+    /**
+     * Determines whether the action should be applied to data streams
+     */
```

Comment:
Suggestion: I would work on the naming and/or javadoc for this to make it clear that such requests over wildcards are meant to (not) expand to data stream names too.